Source/WTF/wtf/text/CString.cpp

```diff
@@ -169,4 +169,30 @@ bool CStringHash::equal(const CString& a, const CString& b)
     return a == b;
 }
 
+enum class ASCIICase { Lower, Upper };
+
+template<ASCIICase type>
+CString convertASCIICase(std::span<const char8_t> input)
+{
+    if (input.empty())
+        return CString(""_s);
+
+    std::span<char> characters;
+    auto result = CString::newUninitialized(input.size(), characters);
+    size_t i = 0;
+    for (auto character : input)
+        characters[i++] = byteCast<char>(type == ASCIICase::Lower ? toASCIILower(character) : toASCIIUpper(character));
+    return result;
```

Comment:
```suggestion
    std::span<char8_t> characters;
    auto result = CString::newUninitialized(input.size(), characters);
    size_t i = 0;
    for (auto character : input)
        characters[i++] = type == ASCIICase::Lower ? toASCIILower(character) : toASCIIUpper(character);
    return byteCast<char>(result);
```

Reply:
I think what you suggest is not possible because of `error: non-const lvalue reference to type 'span<char>' cannot bind to a value of unrelated type 'span<char8_t>'`. I'll keep this as it was.

Reply:
Makes sense; my suggestion was wrong.

Do we need the `byteCast<char>`? I’m pretty sure a `char` can be assigned to a `char8_t` without a cast.